tests/components/habitica/test_config_flow.py

```diff
@@ -113,9 +113,13 @@ async def test_manual_flow_config_exist(hass, aioclient_mock):
     assert result["type"] == "form"
     assert result["step_id"] == "user"
 
+    mock_obj = MagicMock()
+    mock_obj.user.get.return_value = Future()
+    mock_obj.user.get.return_value.set_result({"api_user": "test-api-user"})
```

Comment:
If we autospec the mock when patching the client the mock will be created with methods that are coroutines when appropriate by default. We can then set the return value of the specced method.

Reply:
@MartinHjelmare , the autospec is a very new concept for me - sorry, I wasn't able to make it work. 
Seems like underlying `HabitipyAsync` client is constructing it's methods and parameters dynamically, which makes it confusing to autospec.

My latest try was like:
```python
    with patch(
        "homeassistant.components.habitica.config_flow.HabitipyAsync",
        autospec=True
    ) as mock_client:
        mock_client.return_value.user.return_value.get.return_value = {"api_user": "test-api-user"}
```
is ending up in the error `AttributeError: Mock object has no attribute 'user'`

Reply:
Ok. See another approach below.

Reply:
That's some great stuff! Thank you

Reply:
Learning a lot with all these structures and practises, thanks 💥